.aws/src/main.ts

```diff
@@ -114,18 +116,88 @@ class ListAPI extends TerraformStack {
+        skipFinalSnapshot: true,
+        engine: 'aurora-mysql',
+        engineMode: 'serverless',
+        scalingConfiguration: {
+          minCapacity: config.rds.minCapacity,
+          maxCapacity: config.rds.maxCapacity,
+          autoPause: false,
+        },
+        deletionProtection: false,
+      },
+      tags: config.tags,
+    });
+  }
+
   private createPocketAlbApplication(dependencies: {
     pagerDuty: PocketPagerDuty;
     region: datasources.DataAwsRegion;
     caller: datasources.DataAwsCallerIdentity;
     secretsManagerKmsAlias: kms.DataAwsKmsAlias;
     snsTopic: sns.DataAwsSnsTopic;
+    vpc: PocketVPC;
   }): PocketALBApplication {
-    const { pagerDuty, region, caller, secretsManagerKmsAlias, snsTopic } =
+    const { pagerDuty, region, caller, secretsManagerKmsAlias, snsTopic, vpc } =
       dependencies;
 
     const databaseSecretsArn = `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:${config.name}/${config.environment}/READITLA_DB`;
 
+    /**
+     * Create an RDS instance if we are working in the Dev account.
+     * This is only to facilitate testing
+     */
+    let rdsCluster: ApplicationRDSCluster;
+
+    const secretResources = [
+      `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:Shared`,
+      `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:Shared/*`,
+      secretsManagerKmsAlias.targetKeyArn,
+      `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:${config.name}/${config.environment}`,
+      `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:${config.name}/${config.environment}/*`,
+      `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:${config.prefix}`,
+      `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:${config.prefix}/*`,
+    ];
+
+    // Set out the DB connection details for the production (legacy) database.
+    let databaseSecretEnvVars = {
+      readHost: `${databaseSecretsArn}:read_host::`,
+      readUser: `${databaseSecretsArn}:read_username::`,
+      readPassword: `${databaseSecretsArn}:read_password::`,
+      writeHost: `${databaseSecretsArn}:write_host::`,
+      writeUser: `${databaseSecretsArn}:write_username::`,
+      writePassword: `${databaseSecretsArn}:write_password::`,
+    };
+
+    if (config.isDev) {
+      rdsCluster = this.createRds(vpc);
+      // Add Dev RDS-specific secrets if in Dev environment
+      secretResources.push(rdsCluster.secretARN);
+
+      // Specify DB connection details for the RDS cluster on Dev
+      databaseSecretEnvVars = {
+        readHost: rdsCluster.secretARN + ':host::',
+        readUser: rdsCluster.secretARN + ':username::',
+        readPassword: rdsCluster.secretARN + ':password::',
+        writeHost: rdsCluster.secretARN + ':host::',
+        writeUser: rdsCluster.secretARN + ':username::',
+        writePassword: rdsCluster.secretARN + ':password::',
+      };
+    }
```

Comment:
So the last remaining issues with deploying to Dev were: 

1. The fact that no secret for `READITLA_DB` existed in the dev AWS account - fixed during pairing with @kkyeboah earlier today.
2. The fact that the RDS cluster needed to go to its own secret to retrieve DB connection params and not the `READITLA_DB` one - fixed over a few trial & error commits.

Note that this PR shows file changes from a couple of merge commits that are not related to this PR.

Reply:
I ran into this too :) thanks for fixing!